coalib/parsing/CliParsing.py

```diff
@@ -87,3 +87,31 @@ def parse_custom_settings(sections,
                                origin=origin,
                                section_name=key_touple[0],
                                from_cli=True)
+
+
+def check_conflicts(cli_sections, log_printer):
+    '''
+    Checks if there are any conflicting aruments passed
+
+    :return:            True if no conflicts
+    :raises SystemExit: If there are conflicting arguments (exit code: 2)
+    '''
+    conflicts = defaultdict(set)
+
+    # Add any conflict definitions here
+    conflicts["no_config"].add("save")
+    conflicts["no_config"].add("find_config")
+
+    args = list(conflicts.keys())
+    for arg in args:
+        for conflict in conflicts[arg]:
+            conflicts[conflict].add(arg)
+
+    for section in cli_sections:
+        for arg in cli_sections[section]:
+            for conflict in conflicts[arg]:
```

Comment:
There are nicer ways of doing this because you use `defaultdict(set)`
`conflicts[arg] & set(cli_sections[section])` is a simpler way which uses the fact that you have sets rather than a for loop and `if` which would be slower.

I think these nested for loops can be optimized.